pipeline_dp/budget_accounting.py

```diff
@@ -100,7 +101,188 @@ def compute_budgets(self):
         for requested_budget in self._requested_budgets:
             eps = delta = 0
             if total_weight_eps:
-                eps = requested_budget.use_eps * self._eps * requested_budget.weight / total_weight_eps
+                numerator = requested_budget.use_eps * self._eps * requested_budget.weight
+                eps = numerator / total_weight_eps
             if total_weight_delta:
-                delta = requested_budget.use_delta * self._delta * requested_budget.weight / total_weight_delta
+                numerator = requested_budget.use_delta * self._delta * requested_budget.weight
+                delta = numerator / total_weight_delta
             requested_budget.budget.set_eps_delta(eps, delta)
+
+
+@dataclass
+class MechanismSpec:
+    """Specifies the parameters for a mechanism.
+
+    NoiseKind defines the kind of noise distribution.
+    noise is the minimized noise standard deviation.
+    """
+    noise_kind: NoiseKind
+    _noise_standard_deviation: float = None
+
+    @property
+    def noise(self):
+        """Noise value for the mechanism.
+
+        Raises:
+            AssertionError: The noise value is not calculated yet.
+        """
+        if self._noise_standard_deviation is None:
+            raise AssertionError(
+                "Noise standard deviation is not calculated yet.")
+        return self._noise_standard_deviation
+
+
+@dataclass
+class MechanismSpecInternal:
+    """Stores sensitivity and weight not exposed in MechanismSpec."""
+    sensitivity: float = 1
+    weight: float = 1
+    mechanism_spec: MechanismSpec = None
+
+
+class PLDBudgetAccountant:
+    """Manages the privacy budget for privacy loss distributions.
+
+    It manages the privacy budget for the pipeline using the
+    Privacy Loss Distribution (PLD) implementation from Google's
+    dp_accounting library.
+    """
+
+    def __init__(self,
+                 total_epsilon: float,
+                 total_delta: float,
+                 pld_discretization: float = 1e-4):
+        """Constructs a PLDBudgetAccountant.
+
+        Args:
+            total_epsilon: epsilon for the entire pipeline.
+            total_delta: delta for the entire pipeline.
+            pld_discretization: `value_discretization_interval` in PLD library.
+                Smaller interval results in better accuracy, but increases running time.
+
+        Raises:
+            ValueError: Arguments are missing or out of range.
+        """
+
+        _validate_epsilon_delta(total_epsilon, total_delta)
+
+        self._total_epsilon = total_epsilon
+        self._total_delta = total_delta
+        self._mechanisms = []
+        self.minimum_noise_std = None
+        self._pld_discretization = pld_discretization
+
+    def request_budget(self, noise_kind: NoiseKind) -> MechanismSpec:
```

Comment:
Please add sensitivity and weight as arguments (with default values = 1) and pass them to `MechanismSpecInternal`